### DISCUSSION TITLE
VT132 and DRCS support

### DISCUSSION THREAD

**Discussion by Frlee42** [thehighnibble/thehighnibble.github.io#42] [VT132 terminal emulation]
I understood that the latest VT132 firmware supports DRCS support (downloadable fonts). However, when I ask for a device report, using a primary DA exchange ( "CSI c" or "CSI  0 C"), I got the answer [?64;1;6;22c This means : Terminal Type 64, 132 column(1), selective erase(6) and 22 (no clue) , I'm missing '7', which indicates that the terminal has DRCS support. (see https://vt100.net/docs/vt220-rm/chapter4.html#T4-2 for a in-depth explanation).
It is maybe because DRCS is a kind of VT132 backport from the VT220 specs, the '7' is therefor omitted or is this a bug?

**Comment by dmcnaugh** [comment]
> It is maybe because DRCS is a kind of VT132 backport from the VT220 specs, the '7' is therefor omitted or is this a bug?

Sort of the first suggestion, but the second also applies.
I've never looked at the DA response or thought about updating it as I've added functionality.

The response is hard coded. I think the 22 is for color support (VT520/525 Programmers Reference Manual)

I should probably add
- '7' for DRCS
- and '9' for 'NRCS'

Is this important & urgent ie. is it stopping you from using the DRCS feature?

**Reply by Frlee42** [reply]
Thanks for your fast reply. I was experimenting with automatic DRCS detection when I discovered this issue. Of course,I can do a manual override, but it would be nice if it works as it supposed to work.
I guess a very limited number of VT132 users will ever use this function, so I wouldn't call this a prio 1 problem request.
But it would be fun if it works. I love new firmwares :)

Another thing I noticed is a number of  non printable ascii characters before the actual [?64:1;6:22c being sent.
I'm not sure if these weird characters are cause by my coding, or that the VT132 sends these erroneously . Another dialog respond (cursor positioning) works correctly in my program though. I can't het rid of these non printables on my screen, which is a bit disturbing. Can you reproduce this and is it a bug or am I doing something wrong?

**Reply by dmcnaugh** [reply]
The response begins with `<Esc> [ ` the DEC CSI. So in full `<ESC>[?64:1;6:22c` 
What other non printable characters are you getting?

**Reply by Frlee42** [reply]
This turned out to be a bug somewhere in my code.. When I did the same DRCS check in a few lines of C code, I got the correct response without the  leading rubbish characters. My appologies

**Reply by Frlee42** [reply]
Do you have a few DRCS files I can download for testing? I tried to generate a DRCS fontfile as described on VT100.NET but I guess I'm doing something wrong, The VT132 does not recognise it and on the 'secret' option on the  SET-UP C screen, the Dscs entry  doesn't change...
